pkg/common/ingress/controller/listers.go

```diff
@@ -121,9 +145,29 @@ func (ic *GenericController) createListers(disableNodeLister bool) (*ingress.Sto
 
 			ic.syncQueue.Enqueue(cur)
 		},
-	}
+	})
+	lister.Ingress.Lister, controller.Ingress = ingressInformer.Lister(), ingressInformer.Informer()
 
-	secrEventHandler := cache.ResourceEventHandlerFuncs{
+	endpointInformer := si.Core().V1().Endpoints()
+	endpointInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			ic.syncQueue.Enqueue(obj)
+		},
+		DeleteFunc: func(obj interface{}) {
+			ic.syncQueue.Enqueue(obj)
+		},
+		UpdateFunc: func(old, cur interface{}) {
+			oep := old.(*apiv1.Endpoints)
+			ocur := cur.(*apiv1.Endpoints)
+			if !reflect.DeepEqual(ocur.Subsets, oep.Subsets) {
+				ic.syncQueue.Enqueue(cur)
+			}
+		},
+	})
+	lister.Endpoint.Lister, controller.Endpoint = endpointInformer.Lister(), endpointInformer.Informer()
```

Comment:
Move the endpoint informer declaration between the `lister.Secret.Lister` assignment and the cm informer declaration in order to reduce the diff.